motr/idx_dix.c

```diff
@@ -978,9 +987,11 @@ static void dix_index_create_ast(struct m0_sm_group *grp, struct m0_sm_ast *ast)
 	struct m0_dix_req       *dreq = &dix_req->idr_dreq;
 	struct m0_dix            dix;
 	int                      rc;
+	struct m0_fid		 dix_pver;
 
 	M0_ENTRY();
 	dix_build(oi, &dix);
+	dix_pver = dix_req->idr_oi->oi_idx->in_attr.idx_pver;
```

Comment:
After "dix_build(oi, &dix);", we should use the result "dix".
The "oi" is input of function dix_build(), and "dix" is output.
We should not use "dix_req->idr_oi->oi_idx->in_attr.idx_pver" any more.

Reply:
This change is applicable for the "index create". 
Already we are assigned pver and layout type in idx_op_init() function. 
"struct idx_attr" is available in dix_req due to that reason used dix_req to get pool version. "strut idx_attr" was not available in "dix"
dix_build() is common for all operations and major changes for PUT, GET, DEL, and NEXT operations.

Reply:
```
diff --git a/motr/idx_dix.c b/motr/idx_dix.c
index 1fa39ba2..84b36f9f 100644
--- a/motr/idx_dix.c
+++ b/motr/idx_dix.c
@@ -560,6 +560,18 @@ static void dix_build(const struct m0_op_idx *oi,
                        out->dd_layout.dl_type = DIX_LTYPE_DESCR;
                        out->dd_layout.u.dl_desc.ld_pver = idx->in_attr.idx_pver;
                }
+       } else if (M0_IN(opcode, (M0_EO_CREATE))) {
+               /*
+                * Use default layout for all indices:
+                * - city hash function;
+                * - infinity identity mask (use key as is);
+                * - default pool version (the same as for root index).
+                * In future client user will be able to pass layout as an argument.
+                */
+               dix.dd_layout.dl_type = DIX_LTYPE_DESCR;
+               m0_dix_ldesc_init(&dix.dd_layout.u.dl_desc,
+                                 &(struct m0_ext) { .e_start = 0, .e_end = IMASK_INF },
+                                 1, HASH_FNC_CITY, &dix_inst(oi)->di_index_pver);
        }
 }
 
@@ -976,22 +988,12 @@ static void dix_index_create_ast(struct m0_sm_group *grp, struct m0_sm_ast *ast)
        struct dix_req          *dix_req = ast->sa_datum;
        struct m0_op_idx        *oi = dix_req->idr_oi;
        struct m0_dix_req       *dreq = &dix_req->idr_dreq;
-       struct m0_dix            dix;
+       struct m0_dix            dix = { 0 };
        int                      rc;
 
        M0_ENTRY();
        dix_build(oi, &dix);
-       /*
-        * Use default layout for all indices:
-        * - city hash function;
-        * - infinity identity mask (use key as is);
-        * - default pool version (the same as for root index).
-        * In future client user will be able to pass layout as an argument.
-        */
-       dix.dd_layout.dl_type = DIX_LTYPE_DESCR;
-       m0_dix_ldesc_init(&dix.dd_layout.u.dl_desc,
-                         &(struct m0_ext) { .e_start = 0, .e_end = IMASK_INF },
-                         1, HASH_FNC_CITY, &dix_inst(oi)->di_index_pver);
+
        m0_clink_add(&dreq->dr_sm.sm_chan, &dix_req->idr_clink);
        rc = m0_dix_create(dreq, &dix, 1, NULL, COF_CROW);
        if (rc != 0)
```